### ISSUE TITLE
"final" specifier breaks formatting

### ISSUE THREAD

**Issue by pascal-soveaux** [microsoft/vscode-cpptools#6638]
**Type: LanguageService**

Hello,

Am I on the right place to submit the following issue ?

On my VSCode editor on linux, using the "final" specifier breaks formatting
```   
        class API_INTERNAL IContext /*No final: it works */
        {
            protected:
            virtual ~IContext() = default;
        };

        class API_INTERNAL Context final /*  When using "final": it does not*/
            : public IContext
        {
}
```
Thanks,

Pascal

**Comment by sean-mcmanus** [comment]
I'm not able to repro a formatting problem with your sample code. Can you provide a screenshot? Also, what are your formatting settings?

**Comment by pascal-soveaux** [comment]
I am sorry, I put the blame on the final specifier too quickly. You need to have both attribute and final to reproduce.

You can see that formatting only works if I remove one of the two, with both formatting does not work 
![final](https://user-images.githubusercontent.com/34265976/101545745-ff458580-39a7-11eb-9bee-b3e0435d928d.gif)

The code is wrong (copied/pasted) but it does not changed anything once corrected.

Sorry for that.

Thank you.

Here is what I have in settings
```
"[cpp]": {
    "editor.defaultFormatter": "ms-vscode.cpptools"
  },
  "C_Cpp.formatting": "vcFormat",
  "C_Cpp.commentContinuationPatterns": [
    {
      "start": "/**",
      "continue": "*",
      "end": "*/"
    }
  ],
  "C_Cpp.loggingLevel": "Debug",
  "C_Cpp.default.cppStandard": "c++14",
  "C_Cpp.default.cStandard": "c11",
  "C_Cpp.vcFormat.indent.accessSpecifiers": true,
  "C_Cpp.vcFormat.indent.braces": false,
  "C_Cpp.vcFormat.indent.preprocessor": "none",
  "C_Cpp.vcFormat.indent.withinParentheses": "alignToParenthesis",
  "C_Cpp.vcFormat.newLine.beforeOpenBrace.block": "newLine",
  "C_Cpp.inactiveRegionForegroundColor": "gray",
  "C_Cpp.vcFormat.wrap.preserveBlocks": "never",
  "C_Cpp.vcFormat.space.beforeBlockOpenBrace": false,
  "C_Cpp.vcFormat.newLine.scopeBracesOnSeparateLines": true,
  ```

Version: 1.51.1
Commit: e5a624b788d92b8d34d1392e4c4d9789406efe8f
Date: 2020-11-10T23:31:29.624Z
Electron: 9.3.3
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 5.4.0-56-generic

**Comment by sean-mcmanus** [comment]
Thanks for the info. I've filed a bug on VS with our shared formatting code: https://developercommunity.visualstudio.com/content/problem/1281567/cc-formatting-doesnt-work-with-when-a-class-defini.html

**Comment by sean-mcmanus** [comment]
The fix is available with 1.13.3 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.3
